app/src/androidTest/java/com/github/sdpcoachme/CoachMeTestApplication.kt

```diff
@@ -32,13 +38,26 @@ class CoachMeTestApplication : CoachMeApplication() {
             // Ignore
         }
          */
-        database = MockDatabase()
+        store = CachingStore(MockDatabase(), dataStore, this)
         locationProvider = MockLocationProvider()
-
         // Might be necessary to initialize Places SDK, but for now, we don't need it.
     }
     override val authenticator: Authenticator = MockAuthenticator()
 
+    override fun onTerminate() {
```

Comment:
The naming of superOnTerminate is a bit confusing, maybe you could add a small comment.